anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -559,6 +563,424 @@ def group_by_name(self, group_name):
+        :type active: Optional[bool], defaults to None
+        :return: sqlalchemy query object
+        :rtype: Query
+        """
+        results = db.query(GrypeDBMetadata)
+        if checksum:
+            results = results.filter(GrypeDBMetadata.checksum == checksum)
+        if not isinstance(active, type(None)):
+            results = results.filter(GrypeDBMetadata.active == active)
+        return results
+
+    def record_count(self, group_name: str, db: Session) -> int:
+        """
+        Returns number of records present in the database for a given group (GrypeDBMetadata records)
+
+        :param group_name: name of the group
+        :type group_name: str
+        :param db: sqlaclhemy database session
+        :type db: Session
+        :return: number of records
+        :rtype: int
+        """
+        try:
+            return self._find_match(db).count()
+        except Exception:
+            logger.exception(
+                "Error getting feed data group record count in package feed for group: {}".format(
+                    group_name
+                )
+            )
+            raise
+
+    def _flush_group(
+        self, group_obj: FeedGroupMetadata, operation_id: Optional[str] = None
+    ) -> None:
+        """
+        Flush a specific data group. Do a db flush, but not a commit at the end to keep the transaction open.
+
+        :param group_obj: feed group metadata record
+        :type group_obj: FeedGroupMetadata
+        :param operation_id: UUID4 hexadecimal string
+        :type operation_id:  Optional[str], defaults to None
+        """
+        db = get_session()
+        catalog_client = self._catalog_client
+
+        logger.info(
+            log_msg_ctx(
+                operation_id,
+                group_obj.name,
+                group_obj.feed_name,
+                "Flushing group records",
+            )
+        )
+
+        records = self._find_match(db)
+        for record in records.all():
+            catalog_client.delete_document(record.group_name, record.checksum)
+        records.delete(synchronize_session="evaluate")
+        group_obj.last_sync = None  # Null the update timestamp to reflect the flush
+        group_obj.count = 0
+        db.flush()
+
+    def _enqueue_refresh_tasks(self, db: Session) -> None:
+        """
+        Places a refresh task on the simple queue service for every image id in the image table.
+
+        :param db: sqlalchemy database session
+        :type db: Session
+        """
+        q_client = internal_client_for(SimpleQueueClient, None)
+        subscription_type = "refresh_tasks"
+        image_ids = db.query(Image.id).all()
+        errors = []
+        for result in image_ids:
+            try:
+                task_body = {"image_id": result.id}
+                # if not q_client.is_inqueue(subscription_type, task_body):
+                q_client.enqueue(subscription_type, task_body)
+            except Exception as err:
```

Comment:
Yes because of the behavior mentioned in the above post. All of the code in anchore_engine/clients/services/http.py needs to be overhauled, but that will need to come out of a different PR IMO.

Reply:
resolved

Reply:
https://github.com/anchore/anchore-engine/issues/994